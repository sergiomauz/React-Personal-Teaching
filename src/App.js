/* eslint-disable react/forbid-prop-types */
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import PageContainer from './components/PageContainer';

const App = props => {
  const { store } = props;

  return (
    <BrowserRouter>
      <Provider store={store}>
        <PageContainer />
      </Provider>
    </BrowserRouter>
  );
};

App.propTypes = {
  store: PropTypes.any.isRequired,
};

export default App;
