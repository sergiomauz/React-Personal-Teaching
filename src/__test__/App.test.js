import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import React from 'react';
import App from '../App';
import generateStore from '../redux/store';
import findByTestAttribute from '../helpers/utils';
import {
  SIGN_IN_REQUEST, GET_SESSION, SIGN_OUT,
  GET_USERS_LIST, GET_USER_INFO, GET_MY_PROFILE,
  ADD_USER, UPDATE_USER, UPDATE_USER_N_REDIRECT, REMOVE_USER,
} from '../redux/actions/types';

configure({
  adapter: new Adapter(),
});

const setUp = (initialState = {}) => {
  const testStore = generateStore();
  testStore.dispatch(initialState);
  const wrapper = mount(<App store={testStore} />);
  setTimeout(5000);
  return wrapper;
};

describe('Tests for App component', () => {
  test('Test SIGN_IN_REQUEST', () => {
    const initialState = {
      type: SIGN_IN_REQUEST,
      payload: {
        signedIn: true,
      },
    };
    const appComponent = setUp(initialState);

    const x = findByTestAttribute(appComponent, 'sidebarMenu');
    expect(toJson(x)).toMatchSnapshot();
  });

  // let appComponent;
  // beforeEach(() => {
  //   appComponent = setUp();
  // });
});
