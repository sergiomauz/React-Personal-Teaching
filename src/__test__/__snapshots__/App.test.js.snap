// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for App component Test SIGN_IN_REQUEST 1`] = `
<ul
  className="list-group"
  data-test="sidebarMenu"
>
  <li
    className="list-group-item p-0"
  >
    <Link
      className="list-group-item-action"
      to="/signin"
    >
      <LinkAnchor
        className="list-group-item-action"
        href="/signin"
        navigate={[Function]}
      >
        <a
          className="list-group-item-action"
          href="/signin"
          onClick={[Function]}
        >
          Sign In
        </a>
      </LinkAnchor>
    </Link>
  </li>
  <li
    className="list-group-item p-0"
  >
    <Link
      className="list-group-item-action"
      to="/signup"
    >
      <LinkAnchor
        className="list-group-item-action"
        href="/signup"
        navigate={[Function]}
      >
        <a
          className="list-group-item-action"
          href="/signup"
          onClick={[Function]}
        >
          Sign Up
        </a>
      </LinkAnchor>
    </Link>
  </li>
</ul>
`;
